File: src/apps/Andriod/app/src/main/java/com/dkellycollins/triggerfinger/managers/entity/IPlayerEntityManager.java
package com.dkellycollins.triggerfinger.managers.entity;

import com.dkellycollins.triggerfinger.data.entity.IPlayer;
import com.dkellycollins.triggerfinger.data.model.IVector;

public interface IPlayerEntityManager {

    Iterable<IPlayer> retrieve();
    IPlayer retrieve(int playerId);
    IPlayer retrievePlayerOne();

    int create(IVector position);

    void delete(int playerId);
}

File: src/apps/Android-Kotlin/app/src/main/kotlin/com/dkellycollins/triggerfinger/managers/entity/IPlayerEntityManager.kt
package com.dkellycollins.triggerfinger.managers.entity

import com.dkellycollins.triggerfinger.data.entity.IPlayer
import com.dkellycollins.triggerfinger.data.model.IVector

interface IPlayerEntityManager {

    fun retrieve(): Iterable<IPlayer>
    fun retrieve(playerId: Int): IPlayer
    fun retrievePlayerOne(): IPlayer

    fun create(position: IVector): Int

    fun update(playerId: Int, score: Int, health: Byte)

    fun delete(playerId: Int)
}
